CIE-5514 - Sensitive endpoints to be blocked

raghuvamsi-d opened the pull request.

raghuvamsi-d committed:
heapdump

bbyreddy commented on content/edge/edge-configuration-bundle/configuring-nginx.md:
```diff
@@ -0,0 +1,30 @@
+
+### Blocking Sensitive Endpoints in NGINX
+
+To prevent access to the /service/<service>/heapdump endpoint, create an NGINX location rule that returns 403 Forbidden for all heapdump requests.
```
```suggestion
To prevent access to the `/service/<service>/heapdump` endpoint, create an NGINX location rule that returns "403 Forbidden" for all heapdump requests.
```

raghuvamsi-d committed:
Update content/edge/edge-configuration-bundle/configuring-nginx.md

Co-authored-by: Bhaskar Reddy Byreddy <bhaskar.byreddy@cumulocity.com>

raghuvamsi-d committed:
root user highlight

raghuvamsi-d committed:
Merge branch 'doc/edge-vm-security-fix' of https://github.com/Cumulocity-IoT/c8y-docs into doc/edge-vm-security-fix

rthrippleton commented on content/edge/edge-configuration-bundle/configuring-nginx.md:
This will also need a release note referring to this, pointing out that it's actually something important

raghuvamsi-d replied:
Sure, but this is just a doc change and no actual release, would it still be ok to add a release note since saran is already preparing a communication which includes this too.

rthrippleton replied:
You can have changelogs/release notes that aren't a product change, but just an "Announcement", which would fit here. Having it as an announcement makes it clear that the upcoming communication to customers definitely applies even to Edge.

raghuvamsi-d replied:
Taken note of this suggestion and will be implemented for later scenarios, this, however, is an exception and is excluded from the release notes for this instance. Thank you @rthrippleton

raghuvamsi-d committed:
co-pilot suggestions

BeateRixen commented on content/edge/edge-configuration-bundle/configuring-nginx.md:
```diff
@@ -0,0 +1,30 @@
+---
+weight: 50
+title: Configuring Nginx
```
```suggestion
title: Configuring NGINX
```

raghuvamsi-d committed:
making it generic

raghuvamsi-d committed:
Update content/edge/edge-configuration-bundle/configuring-nginx.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy commented on content/edge/edge-configuration-bundle/configuring-nginx.md:
````diff
@@ -0,0 +1,40 @@
+The following example shows how to block a specific endpoint. You can modify the location pattern to match any endpoint you want to disable:
+
+```shell
+CONFIG_FILE="/etc/nginx/conf.d/blocked_endpoints.common.location.conf"
````
Does the file name have any significance? If not, can we rename it to blocked-endpoints.conf?

raghuvamsi-d replied:
Significance in the way we are importing the files into post.nginx.conf file.

```
http {
    include preamble.http.conf;

    server {
        include http_listen.conf;
        include conf.d/*.common.location.conf;
        include conf.d/*.http.location.conf;

        # Temporary (hopefully) backward compatibility for DataHub
        include conf.d/common/*.location.conf;
    }

    server {
        include https_listen.conf;
        include conf.d/*.common.location.conf;
        include conf.d/*.https.location.conf;
        include conf.d/installation/*.location.post.conf;
    }

    include conf.d/*.http.conf;
}

```


Any combination works here as long as the regex can parse the new file name.

bbyreddy replied:
ok, then, let us name the file as `blocked-endpoints.common.location.conf`

raghuvamsi-d replied:
Done, changed as suggested

bbyreddy commented on content/edge/edge-configuration-bundle/configuring-nginx.md:
```diff
@@ -0,0 +1,40 @@
+
+# Add your exact regex inside the location directive
+# Example: 
+# location ~* ^/service/[^/]+/<endpoint>$ {
```
Could you put an example service endpoint in this?

raghuvamsi-d replied:
Done, Added few examples

raghuvamsi-d committed:
adding more examples

raghuvamsi-d committed:
Merge branch 'doc/edge-vm-security-fix' of https://github.com/Cumulocity-IoT/c8y-docs into doc/edge-vm-security-fix

raghuvamsi-d committed:
rename file

raghuvamsi-d committed:
restric specific tokens

raghuvamsi-d committed:
comment correction

raghuvamsi-d committed:
correcting the token type in docs

raghuvamsi-d committed:
basic auth

bbyreddy commented on content/edge/edge-configuration-bundle/configuring-nginx.md:
```diff
@@ -0,0 +1,69 @@
+
+### Blocking sensitive endpoints in NGINX
+
+To block access to specific endpoints, create an NGINX location rule that returns "403 Forbidden" for matching requests. You can use this approach to disable any endpoint by configuring the appropriate location pattern.
```
```suggestion
Configure an NGINX `location` block to return a `403 Forbidden` status code for all matching incoming requests. This method can be used to effectively disable access to an endpoint with a defined URL pattern or to blacklist specific users within that location block.
```

raghuvamsi-d committed:
Update content/edge/edge-configuration-bundle/configuring-nginx.md

Co-authored-by: Bhaskar Reddy Byreddy <bhaskar.byreddy@cumulocity.com>

bbyreddy commented on content/edge/edge-configuration-bundle/configuring-nginx.md:
```diff
@@ -0,0 +1,69 @@
+cat >> "$CONFIG_FILE" <<'EOF'
+
+# Add your exact regex inside the location directive
+# Example1: Blocks health endpoint only for cep service
```
Update this comment as it is pointing to the endpoint not the http authorization header.

raghuvamsi-d committed:
Clarify blocking logic in NGINX configuration

Updated comments to clarify blocking requests based on HTTP Authorization header.

raghuvamsi-d committed:
refining doc

raghuvamsi-d committed:
reduce the doc scope
